### PR TITLE
Allow for printing hex strings

### PR DESCRIPTION
**Pull request by valakuzhyk**
This will allow us to start printing hex values! The beginnings of a memory dump.

### CODE REVIEW

File: src/io/terminal.c [hunk 1]
```diff
@@ -76,3 +76,41 @@ void terminal_write(const char* data) {
 		terminal_putchar(data[i]);
 }
 
+// writes the hex representation of a number on the terminal.
+void terminal_write_hex(int n) {
+	if (n < 0) {
+		terminal_putchar('-');
+    	n = -n;
```
Comment (rrogowski) [inline src/io/terminal.c]: fix formatting pls!

File: src/io/terminal.c [hunk 2]
```diff
@@ -76,3 +76,41 @@ void terminal_write(const char* data) {
...
+
+		if (digit < 10) {
+			terminal_putchar('0' + digit);
+		} else {
+			terminal_putchar('A' + digit - 10);
+		}
+
+		n = n % digitBase;
+		digitBase /= 16;
+	} while (n != 0);
```
Comment (rrogowski) [inline src/io/terminal.c]: This results in incorrect behavior. We should do `while (num_digits > 0)` and decrement with each iteration.

**Review by rrogowski — CHANGES_REQUESTED** [review]

File: src/io/terminal.c (hunk 2 above)
Comment (valakuzhyk) [reply src/io/terminal.c]: Fixed.

File: src/io/terminal.c (hunk 1 above)
Comment (valakuzhyk) [reply src/io/terminal.c]: Fixed.

**Review by rrogowski — APPROVED** [review]

**Comment by rrogowski** [comment]
LGTM

**Pull request merged**